File: iut/TP5_TP6/indicatrice.py
def premavcn(k, n):
    while (n != 0):  
        tmp = k % n
        k = n
        n = tmp
    if (k == 1):
        return 1
    return 0

def indicatrice(n):
    k = 1
    res = 0
    while (k <= n):
    
        if (premavcn(k, n)):
            res+=1
        k+=1
    
    return res


n = int(input(("N : ")))
x = indicatrice(n)
print(f"Le resultat est : {x}")

File: iut/TP5_TP6/indicatrice.c
#include <stdio.h>

int premavcn(int k, int n)
{
    int tmp;
    while (n != 0)
    {
        tmp = k % n;
        k = n;
        n = tmp;   
    }
    if (k == 1)
        return 1;
    return 0;
}

int indicatrice(int n)
{
    int res; 
    int k;

    k = 1;
    res = 0;
    while (k <= n)
    {
        if (premavcn(k, n))
            res++;
        k++;
    }
    return res;
}

int main(void)
{
    int n;
    printf("N : ");
    scanf("%i", &n);
    printf("Le resultat est : %i",indicatrice(n));
    return 0;
}
